source/extensions/filters/common/rbac/matchers/upstream_ip.cc

```diff
@@ -0,0 +1,56 @@
+#include "source/extensions/filters/common/rbac/matchers/upstream_ip.h"
+
+#include "envoy/config/core/v3/extension.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/stream_info/upstream_address_set.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace RBAC {
+namespace Matchers {
+
+using namespace Filters::Common::RBAC;
+
+bool UpstreamIpMatcher::matches(const Network::Connection&, const Envoy::Http::RequestHeaderMap&,
+                                const StreamInfo::StreamInfo& info) const {
+
+  if (!info.filterState().hasDataWithName(StreamInfo::UpstreamAddressSet::key())) {
+    ENVOY_LOG(warn,
```

Comment:
I think this probably should't be a warn as this will log spam at very high rate at misconfig. I would either use a stat or if you want to see this in the log I might use something like `ENVOY_LOG_EVERY_POW_2_TO_LOGGER` (or see other similar macros).